### ISSUE TITLE
Highlight authorship through inline colors for all users

### ISSUE THREAD

**Issue by nidico** [codimd/react-client#608]
[Component: Editor UI]

## Status quo / problem:

- Authorship information of CodiMD editor content is currently displayed through a vertical colored bar in the line number gutter besides the editor (for the initial author of a line) and by horizontally underlining characters (for edits done by other users in the same line).
- The different UI for "initial author of a line" and "other authors" is not particularly intuitive, as there isn't necessarily a relevant difference between "initial author of a line" and "other authors" after a bit of collaborative editing.
- It works different (and - at least for some - worse) than Etherpad, so it might be a migration obstacle.

## Simple proposal (no bars, only underline, "Etherpad-style"):

- Always highlight author through inline coloring (by colored underlining or background).
- Don't show the bar in the gutter anymore.
- This is just like Etherpad.

## Alternatives:

Assuming that the original intention of the "vertical bar feature" in CodiMD is less clutter (by less lines overall), the following alternatives try to keep that promise:

**Alternative 1 (dynamic bars, funny):**

- Color the "bar in the gutter" by the user with the majority of a block (or line) and underline the edits of other users.
- However: This is just a bit more intuitive than the current version. Also it's a bit strange that the coloring would switch during editing.

**Alternative 2 (bars only for uniformly authored blocks/lines):**

- Show (side)bars instead of underlining text for blocks/lines which are authored by only one user.
- To be defined: What is a block?
- Might be worth a try.

**Alternative 3 (bars for single reference user for the entire document, advanced):**

- Like alternative 1, but show the vertical bar on the entire document for a defined reference user and underline edits of other users.
- That user can be selected dynamically (most authored currently visible text) or explicitly (globally or per document or per individual user's client).
- Allowing to switch the reference user within the client might be an interesting UI experiment.

**Alternative 4:**

- Any of the above, toggleable.

## Related:

- Authorship colors are currently only shown for text written by unauthenticated users, but this is [another issue](codimd/server#505).
- Highlight authorship information in view mode (issue pending).

**Comment by nidico** [comment]
@DerMolly - is this really blocked by backend development? Although I don't know much about the CodiMD architecture, I would be surprised if "simple solution - etherpad-style" and most of the alternatives couldn't be implemented purely in the frontend (apart from things like persisting the reference user in alternative 3, which would be completely optional).

Do I overlook something?

**Comment by DerMolly** [comment]
I‘m not sure we can effectively work on this if there is no backend to communicate with especially over websocket.

**Comment by nidico** [comment]
Is the backend API (I assume this is part of the OT stuff) documented? If authorship information is available for each received junk of text, all of the above should be possible (I would say, naively).

**Comment by mrdrogdrog** [comment]
The hole "show who edited this" feature is currently blocked, because we don't have any data, api spec or data structure for this yet
